adr/graph-feature-flags.md

```diff
@@ -0,0 +1,212 @@
+
+### Conflicting Feature Flags that replace the same subgraph are allowed
+
+It's possible to have multiple feature flags that replace the same subgraph.
+This is useful and allowed for testing different variants of the same subgraph.
+
+### Feature Flags that replace the same subgraph are not allowed in a group feature flag
+
+A feature flag group is never allowed to contain multiple feature flags that replace the same subgraph.
+This would result in an ambiguous configuration.
+
+### Limiting a feature flag to specific federated graphs
+
+We've already established a pattern (label matching) to limit a subgraph to specific federated graphs.
+The same pattern can be used to limit a feature flag to specific federated graphs.
+
+```shell
+wgc federated-graph create production --label-matcher env=staging --routing-url https://graph.domain.com/graphql
+wgc ff create users-v2 --label env=staging --routing-url https://users-v2.domain.com
+wgc ff publish users-v2 --schema ./subgraph-users-v2/schema.graphql
+```
+
+### Combinations of feature flags are only possible via group feature flags
+
+In a previous version of Feature Flags, it was possible to automatically combine multiple feature flags.
+This was removed because it would have resulted in a exponential number of possible combinations,
+which would have required us to limit the total number of feature flags to a very low number.
+
+Instead, we require users to explicitly create group feature flags to combine multiple feature flags.
+With 20 active feature flags and 5 groups, we would have 20+5=25 possible combinations.
+If all 20 feature flags were combined, we would have 20*20=400 possible combinations.
+
+Each time a subgraph gets published, the Control Plane is required to check if all feature flags are still valid,
+which means that the number of possible combinations defines the composition check runs required.
+
+### Composition Checks need to be Feature Flag aware
+
+When a subgraph gets published, the Control Plane needs to check if the subgraph is compatible with all active feature flags.
+If errors occur, they need to be reported in such a way that the user can easily identify which feature flag caused the error.
+
+### Feature Flags must not break composition checks
+
+At any time, all existing feature flags must be compatible with all existing subgraphs,
+based on the label matching rules of the subgraphs and feature flags.
+
+### Feature Flag names must be unique across flags and groups
+
+Within the same namespace, feature flags and group feature flags must have unique names.
+It's not allowed to have a group feature flag with the same name as a feature flag and vice versa.
+It's not allowed to have a group feature flag with the same name as another group feature flag.
+It's not allowed to have a feature flag with the same name as another feature flag.
+
+### Feature Flags are namespaced
+
+Feature Flags are namespaced.
+
+### The Router Engine Config is aware of all Feature Flags
```

Comment:
for users to get all the feature flags would it be
```
wgc feature-flags list
``` ?

Reply:
yes